language/move-vm/runtime/src/interpreter.rs

```diff
@@ -311,6 +341,20 @@ impl Interpreter {
         for _ in 0..expected_args {
             args.push_front(self.operand_stack.pop()?);
         }
+
+        if self.runtime_config.paranoid_type_checks {
+            let is_generic = !ty_args.is_empty();
```

Comment:
Nit: you don't have to branch based on whether the function is generic or not. Just call `function.parameter_types()[expected_args - i - 1].subst(&ty_args)?` and it'll work for both generic and non-generic functions -- if `ty_args` is empty, then the substitution is effectively equivalent to a clone.

Reply:
> efficiency of subst

No need to worry about this. `subst`'s implementation is near identical to clone.